### PR TITLE
HMOS RainyView Library

### PR DESCRIPTION
**Pull request by priyankabhadrappanavar**

### CODE REVIEW

File: rainyview/src/main/java/me/samlss/view/RainyView.java [hunk 1]
```diff
@@ -262,16 +258,14 @@ public void onRefreshed(Component component) {
         w = component.getWidth();
```
Comment (samuel-abhishek) [inline rainyview/src/main/java/me/samlss/view/RainyView.java]: variable declaration and initialization can be done in the same statement.

File: rainyview/src/main/java/me/samlss/view/RainyView.java [hunk 2]
```diff
@@ -262,16 +258,14 @@ public void onRefreshed(Component component) {
...
-        int oldheight = 0;
         stop();
 
         mLeftCloudPath.reset();
         mRightCloudPath.reset();
 
         //view's center x coordinate
         float centerX;
-        centerX = w / 2;
+        centerX = w / 2.0f;
```
Comment (samuel-abhishek) [inline rainyview/src/main/java/me/samlss/view/RainyView.java]: variable declaration and initialization can be done in the same statement. Please check all the files, this applies in all the cases wherever applicable.

Comment (samuel-abhishek) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: keep the minimum distance btw variable declaration and usage to 5, to increase the readability of the code. (checkstyle issue)

File: rainyview/src/main/java/me/samlss/view/RainyView.java [hunk 3]
```diff
@@ -439,112 +430,115 @@ public void release() {
...
-     * */
-    private void createRainDrop() {
 
-        if (mRainDrops.size() >= mRainDropMaxNumber || mRainRect.isEmpty()) {
-            return;
-        }
+        /**
+         * Now create a random raindrop.
+         * */
+        private void createRainDrop() {
```
Comment (samuel-abhishek) [inline rainyview/src/main/java/me/samlss/view/RainyView.java]: this method throws IllegalArgumentException in a specific condition, therefore add "throws" clause to the method signature.

File: rainyview/src/main/java/me/samlss/view/RainyView.java [hunk 4]
```diff
@@ -439,112 +430,115 @@ public void release() {
...
+            if (!mRainDrops.isEmpty()) {
+                invalidate();
+            }
         }
 
-        if (!mRainDrops.isEmpty()) {
-            invalidate();
+        @Override
+        public void run() {
+            createRainDrop();
```
Comment (samuel-abhishek) [inline rainyview/src/main/java/me/samlss/view/RainyView.java]: might throw IllegalArgumentException as per the method definition. Surround it with try/catch

File: rainyview/src/main/java/me/samlss/view/RainyView.java [hunk 5]
```diff
@@ -1,358 +1,435 @@
...
-                leftCloudEndX, leftCloudEndY), leftCloudBottomHeight, leftCloudBottomRoundRadius, Path.Direction.CW);
+        mLeftCloudPath.addRoundRect(new RectFloat(leftCloudEndX - leftCloudWidth, leftCloudEndY - leftCloudBottomHeight,
+                leftCloudEndX, leftCloudEndY), leftCloudBottomHeight,
+                leftCloudBottomRoundRadius, Path.Direction.CLOCK_WISE);
 
         float leftCloudTopCenterY = leftCloudEndY - leftCloudBottomHeight;
+
         float leftCloudRightTopCenterX = leftCloudEndX - leftCloudBottomRoundRadius;
+
         float leftCloudLeftTopCenterX  = leftCloudEndX - leftCloudWidth + leftCloudBottomRoundRadius;
```
Comment (samuel-abhishek) [inline rainyview/src/main/java/me/samlss/view/RainyView.java]: Unecessary whitelines/whitespaces. Please remove. Applies to all the files...

File: entry/src/ohosTest/java/me/samlss/rainyviewdemo/ExampleOhosTest.java [hunk 6]
```diff
@@ -0,0 +1,209 @@
+package me.samlss.rainyviewdemo;
```
Comment (samuel-abhishek) [inline entry/src/ohosTest/java/me/samlss/rainyviewdemo/ExampleOhosTest.java:17]: Add the copyright for this file

File: README.md [hunk 7]
```diff
@@ -1,69 +1,67 @@
 # RainyView
-[![Download](https://api.bintray.com/packages/samlss/maven/rainyview/images/download.svg)](https://bintray.com/samlss/maven/rainyview/_latestVersion)   [![Api reqeust](https://img.shields.io/badge/API-11+-brightgreen.svg?style=flat)](https://android-arsenal.com/api?level=11#l11)    [![Apache License 2.0](https://img.shields.io/hexpm/l/plug.svg)](https://github.com/samlss/RainyView/blob/master/LICENSE)  [![Blog](https://img.shields.io/badge/samlss-blog-orange.svg)](https://blog.csdn.net/Samlss)
 
-A rainy rainy rainy view. ( ˘•灬•˘ )
+A HMOS library which provides rainy view animation
```
Comment (samuel-abhishek) [inline README.md:3]: Please add the build & quality report badges

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Please check the review comments and do the required changes

File: rainyview/src/main/java/me/samlss/view/RainyView.java (hunk 1 above)
Comment (priyankabhadrappanavar) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: Done

File: rainyview/src/main/java/me/samlss/view/RainyView.java (hunk 2 above)
Comment (priyankabhadrappanavar) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: Done

File: rainyview/src/main/java/me/samlss/view/RainyView.java (hunk 3 above)
Comment (priyankabhadrappanavar) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: Done

File: rainyview/src/main/java/me/samlss/view/RainyView.java (hunk 4 above)
Comment (priyankabhadrappanavar) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: Done

File: rainyview/src/main/java/me/samlss/view/RainyView.java (hunk 5 above)
Comment (priyankabhadrappanavar) [reply rainyview/src/main/java/me/samlss/view/RainyView.java]: Done

File: entry/src/ohosTest/java/me/samlss/rainyviewdemo/ExampleOhosTest.java (hunk 6 above)
Comment (priyankabhadrappanavar) [reply entry/src/ohosTest/java/me/samlss/rainyviewdemo/ExampleOhosTest.java:17]: Done

File: README.md (hunk 7 above)
Comment (priyankabhadrappanavar) [reply README.md:3]: We will resolve this after the project runs on master branch

**Review by samuel-abhishek — APPROVED** [review]
LGTM.

**Review by sharathks84 — COMMENTED** [review]
Add the current images from HarmonyOS in Readme

**Comment by priyankabhadrappanavar** [comment]
> Add the current images from HarmonyOS in Readme
The current gif is from HarmonyOs itself. As the tablet was not available, I have captured the video from the tv cropped it, and then uploaded it.

**Review by sharathks84 — APPROVED** [review]
Okay

**Comment by superlib-framework** [comment]
Ok

**Review by Gleelakrishna — APPROVED** [review]
OK

**Pull request merged**